### PR TITLE
Automated cherry pick of #65412: skip nic that are in failing state

### PR DESCRIPTION
**Pull request by feiskyer**
Cherry pick of #65412 on release-1.9.

#65412: skip nic that are in failing state

### CODE REVIEW

**Comment by feiskyer** [comment]
/sig azure
/kind bug

cc @khenidak @brendandburns @andyzhangx

**Comment by feiskyer** [comment]
ping @andyzhangx

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Comment by feiskyer** [comment]
ping @colemickens @brendandburns for approval

**Comment by mbohlool** [comment]
/hold

1.9 branch is frozen for the release.

**Comment by feiskyer** [comment]
ping @colemickens @brendandburns for approval

**Comment by feiskyer** [comment]
ping @jdumars @brendandburns for approval

File: pkg/cloudprovider/providers/azure/azure_util.go [hunk 1]
```diff
@@ -52,7 +52,8 @@ const (
 	InternalLoadBalancerNameSuffix = "-internal"
 
 	// nodeLabelRole specifies the role of a node
-	nodeLabelRole = "kubernetes.io/role"
+	nodeLabelRole  = "kubernetes.io/role"
```
Comment (jdumars) [inline pkg/cloudprovider/providers/azure/azure_util.go]: nit: stray space

**Comment by jdumars** [comment]
added cherry-pick-candidate to parent PR

**Comment by jdumars** [comment]
@mbohlool PTAL

**Comment by nikhita** [comment]
Closing and reopening to retrigger prow to remove the `do-not-merge/cherry-pick-not-approved` label.

/assign

**Comment by nikhita** [comment]
/unassign

The labels are fixed now. :+1:

**Pull request merged**